src/components/protocol_handler/test/protocol_packet_test.cc

```diff
@@ -188,14 +210,45 @@ TEST_F(ProtocolPacketTest, DeserializeZeroPacket) {
 
 TEST_F(ProtocolPacketTest, DeserializeNonZeroPacket) {
   // Set header, serviceType, frameData, sessionId
-  uint8_t session_id = 1u;
-  uint8_t some_message[] = {0x21, 0x07, 0x02, session_id};
+  const uint8_t session_id = 1u;
+  const uint8_t version_frame_type = 0x21;
+  uint8_t some_message[] = {
+      version_frame_type, kRpc, FRAME_DATA_START_SERVICE_ACK, session_id};
   ProtocolPacket protocol_packet;
   RESULT_CODE res =
       protocol_packet.deserializePacket(some_message, PROTOCOL_HEADER_V2_SIZE);
   EXPECT_EQ(RESULT_OK, res);
 }
 
+TEST_F(ProtocolPacketTest, DeserializePacket_FrameTypeFirst_ResultOK) {
+  // Arrange
+  const uint8_t session_id = 1u;
+  const uint8_t data_size = 1u;
+  const uint8_t version_frame_type = 0x22;
+  // Set protol version - 2 and frame type - first
+  uint8_t message[] = {version_frame_type,
+                       kRpc,
+                       FRAME_DATA_FIRST,
+                       session_id,
+                       0x00,
```

Comment:
@AByzhynar 
I know I'm tedious, but I would suggest to use variables instead of magic.